src/mantle/mantle_query_sync.c

```diff
@@ -75,3 +76,133 @@ GR_RESULT grWaitForFences(
         return GR_ERROR_OUT_OF_MEMORY;
     }
 }
+
+
+GR_RESULT grGetFenceStatus(
+    GR_FENCE fence)
+{
+    GrFence* grFence = (GrFence*)fence;
+    if (fence == NULL){
+	return GR_ERROR_INVALID_POINTER;
+    }
+
+    VkResult res = vki.vkGetFenceStatus(grFence->device, grFence->fence);
+    switch (res){
+    case VK_SUCCESS:
+	return GR_SUCCESS;
+    case VK_NOT_READY:
+	return GR_NOT_READY;
+    case VK_ERROR_DEVICE_LOST:
+	return GR_ERROR_DEVICE_LOST;
+    default:
+	break;
+    }
+    return GR_ERROR_UNKNOWN;
```

Comment:
all functions should return success at the end. return early when an error is hit

Reply:
fixed, but in case of failure in mantle_wsi.c errors will be returned after GR_SUCCESS block (did it to move object destruction to one place)